vispy/visuals/polygon.py

```diff
@@ -38,17 +39,40 @@ class PolygonVisual(Visual):
     """
     def __init__(self, pos=None, color='black',
                  border_color=None, border_width=1, **kwargs):
-        super(PolygonVisual, self).__init__(**kwargs)
-
-        self.mesh = MeshVisual()
-        self.border = LineVisual()
+        self._mesh = MeshVisual()
+        self._border = LineVisual()
         self._pos = pos
         self._color = Color(color)
         self._border_width = border_width
         self._border_color = Color(border_color)
+
         self._update()
-        #glopts = kwargs.pop('gl_options', 'translucent')
-        #self.set_gl_options(glopts)
+        CompoundVisual.__init__(self, [self._mesh, self._border], **kwargs)
+        self._mesh.set_gl_state(polygon_offset_fill=True,
```

Comment:
I don't understand how this change makes it work. It shouldn't as far as I understand `polygon_offset_fill`, because the `border` is drawn _after_ the `mesh`. So, setting `polygon_offset_fill` on `border` should have worked.